app/models/course/assessment/submission.rb

```diff
@@ -91,6 +91,22 @@ def auto_grade!
     AutoGradingJob.perform_later(self)
   end
 
+  # Return the maximum step that the current submission can reach.
+  #
+  # @return [Integer] the maximum step.
+  def max_step
```

Comment:
How will you suggestion to use next_unanswered_question play out?

I think calling it unanswered is fine. Then "steps" are purely a view concept.